cli/args/lockfile.rs

```diff
@@ -1,602 +1,84 @@
-        "foo".to_string(),
-        "shasum".to_string(),
-        Some("sha512-MqBkQh/OHTS2egovRtLk45wEyNXwF+cokD+1YPf9u5VfJiRdAiRwB2froX5Co9Rh20xs4siNPm8naNotSD6RBw==".to_string()),
-      ),
-      dependencies: HashMap::new(),
-    };
-    let check_ok = lockfile.check_or_insert_npm_package(&npm_package);
-    assert!(check_ok.is_ok());
-
-    let npm_package = NpmResolutionPackage {
-      id: NpmPackageId {
-        name: "picocolors".to_string(),
-        version: NpmVersion::parse("1.0.0").unwrap(),
-        peer_dependencies: Vec::new(),
-      },
-      copy_index: 0,
-      dist: NpmPackageVersionDistInfo::new(
-        "foo".to_string(),
-        "shasum".to_string(),
-        Some("sha512-1fygroTLlHu66zi26VoTDv8yRgm0Fccecssto+MhsZ0D/DGW2sm8E8AjW7NU5VVTRt5GxbeZ5qBuJr+HyLYkjQ==".to_string()),
-      ),
-      dependencies: HashMap::new(),
-    };
-    // Integrity is borked in the loaded lockfile
-    let check_err = lockfile.check_or_insert_npm_package(&npm_package);
-    assert!(check_err.is_err());
-
-    let npm_package = NpmResolutionPackage {
-      id: NpmPackageId {
-        name: "source-map-js".to_string(),
-        version: NpmVersion::parse("1.0.2").unwrap(),
-        peer_dependencies: Vec::new(),
-      },
-      copy_index: 0,
-      dist: NpmPackageVersionDistInfo::new(
-        "foo".to_string(),
-        "foo".to_string(),
-        Some("sha512-R0XvVJ9WusLiqTCEiGCmICCMplcCkIwwR11mOSD9CR5u+IXYdiseeEuXCVAjS54zqwkLcPNnmU4OeJ6tUrWhDw==".to_string()),
-      ),
-      dependencies: HashMap::new(),
-    };
-    // Not present in lockfile yet, should be inserted and check passed.
-    let check_ok = lockfile.check_or_insert_npm_package(&npm_package);
-    assert!(check_ok.is_ok());
-
-    let npm_package = NpmResolutionPackage {
-      id: NpmPackageId {
-        name: "source-map-js".to_string(),
-        version: NpmVersion::parse("1.0.2").unwrap(),
-        peer_dependencies: Vec::new(),
-      },
-      copy_index: 0,
-      dist: NpmPackageVersionDistInfo::new(
-        "foo".to_string(),
-        "foo".to_string(),
-        Some("sha512-foobar".to_string()),
-      ),
-      dependencies: HashMap::new(),
-    };
-    // Now present in lockfile, should file due to borked integrity
-    let check_err = lockfile.check_or_insert_npm_package(&npm_package);
-    assert!(check_err.is_err());
+// NOTE(bartlomieju): we don't want a reverse mapping to be possible.
+#[allow(clippy::from_over_into)]
+impl Into<NpmPackageLockfileInfo> for NpmResolutionPackage {
+  fn into(self) -> NpmPackageLockfileInfo {
+    let dependencies = self
+      .dependencies
+      .into_iter()
+      .map(|(name, id)| NpmPackageDependencyLockfileInfo {
+        name,
+        id: id.as_serialized(),
+      })
+      .collect();
+
+    NpmPackageLockfileInfo {
+      display_id: self.id.display(),
+      serialized_id: self.id.as_serialized(),
+      integrity: self.dist.integrity().to_string(),
+      dependencies,
+    }
```

Comment:
Slightly unfortuante that the common case requires cloning now, but not a big deal. We can fix this up when extracting stuff like `NpmPackageId` from the CLI repo.

Reply:
I do agree, ideally we wouldn't require any clones for the lifetime of the lockfile checking